controllers/reconcilers/configuration/prometheus.go

```diff
@@ -122,22 +122,42 @@ func (r *Reconciler) createAdditionalScrapeConfigSecret(cr *v1.Observability, ct
 
 func (r *Reconciler) getOpenshiftMonitoringCredentials(ctx context.Context) (string, string, error) {
 	secret := &kv1.Secret{}
+	osVersionToCompareString := "4.10.0"
+	var isDatasourceV2 bool
+
+	currentOSVersionString, err := utils.GetClusterOSVersion(ctx, r.client)
+	if err != nil {
+		return "", "", err
+	}
+
+	isDatasourceV2, err = utils.HasNewerOrSameClusterVersion(currentOSVersionString, osVersionToCompareString)
+	if err != nil {
+		return "", "", err
+	}
+
+	secretName := "grafana-datasources-v2"
+	if !isDatasourceV2 {
+		secretName = "grafana-datasources"
+	}
 	selector := client.ObjectKey{
 		Namespace: "openshift-monitoring",
-		Name:      "grafana-datasources",
+		Name:      secretName,
 	}
-
-	err := r.client.Get(ctx, selector, secret)
+	err = r.client.Get(ctx, selector, secret)
 	if err != nil {
 		return "", "", err
 	}
 
 	// It says yaml but it's actually json
 	j := secret.Data["prometheus.yaml"]
 
+	type datasourceSecureData struct {
```

Comment:
Does this change still apply pre 4.10 and grafana v2?

Reply:
For pre 4.10 the `basicAuthPassword` is contained in main json body of `grafana-datasources` secret. The `datasource` struct lists `secureJsonData` as optional field. I did test this change on 4.9 cluster and worked as expected